Error while checking build/O2/fullCI for f53042c2605d821cd144d2a85dd614313c24eb20 at 2023-05-29 02:20:
```
## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile reco_ASYNC.log
reco_ASYNC.log:[57998:aod-producer-workflow]: [00:20:25][ERROR] Exception caught: Worker is not authorised to create message with origin(AOD) description(CASCADE_001) subSpec(1) 
[57967:its-tracker]: [00:19:56][ERROR] Too many tracklets per cluster (6.271296), check the detector status and/or the selections.
[57998:aod-producer-workflow]: [00:20:25][ERROR] Exception caught: Worker is not authorised to create message with origin(AOD) description(CASCADE_001) subSpec(1) 
[57998:aod-producer-workflow]: [00:20:27][FATAL] Unhandled o2::framework::runtime_error reached the top of main of o2-aod-producer-workflow, device shutting down. Reason: Worker is not authorised to create message with origin(AOD) description(CASCADE_001) subSpec(1)
[ERROR] pid 57998 (aod-producer-workflow) crashed with or was killed with exit code 1
[ERROR] SEVERE: Device aod-producer-workflow (57998) had at least one message above severity 5: Unhandled o2::framework::runtime_error reached the top of main of o2-aod-producer-workflow, device shutting down. Reason: Worker is not authorised to create message with origin(AOD) description(CASCADE_001) subSpec(1)


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/575a66c48b26bc4e53e5a14612e3f38011d3f40a/slc8_x86-64/o2checkcode/1.0-local580/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11396/f53042c2605d821cd144d2a85dd614313c24eb20/build_O2_fullCI/pretty.html).

--- a/Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h
+++ b/Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h
@@ -53,18 +53,11 @@ namespace o2::aodproducer
 {
 
 template <typename T>
-void addTableToOutput(std::vector<OutputSpec>& o, const std::string& ver = "", header::DataHeader::SubSpecificationType t = 0)
+void addTableToOutput(std::vector<OutputSpec>& o, int t = 0)
 {
-  std::string d = o2::aod::MetadataTrait<T>::metadata::description();
-  if (!ver.empty()) {
-    d += "_";
-    d += ver;
-  }
-  char chDesc[16];
-  std::strcpy(chDesc, d.c_str());
   o.emplace_back(OutputLabel{o2::aod::MetadataTrait<T>::metadata::tableLabel()},
                  o2::aod::MetadataTrait<T>::metadata::origin(),
-                 chDesc,
+                 o2::aod::MetadataTrait<T>::metadata::description(),
                  t,
                  Lifetime::Timeframe);
 }

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -1721,7 +1721,7 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
   auto mcTrackLabelCursor = createTableCursor<o2::aod::McTrackLabels>(pc);
   auto mftTracksCursor = createTableCursor<o2::aod::StoredMFTTracks>(pc);
   auto tracksCursor = createTableCursor<o2::aod::StoredTracksIU>(pc);
-  auto tracksCovCursor = createTableCursor<o2::aod::StoredTracksCov>(pc);
+  auto tracksCovCursor = createTableCursor<o2::aod::StoredTracksCovIU>(pc);
   auto tracksExtraCursor = createTableCursor<o2::aod::StoredTracksExtra>(pc);
   auto ambigTracksCursor = createTableCursor<o2::aod::AmbiguousTracks>(pc);
   auto ambigMFTTracksCursor = createTableCursor<o2::aod::AmbiguousMFTTracks>(pc);
@@ -2680,19 +2680,23 @@ DataProcessorSpec getAODProducerWorkflowSpec(GID::mask_t src, bool enableSV, boo
   using namespace o2::aodproducer;
 
   addTableToOutput<BCs>(outputs);
-  addTableToOutput<Cascades>(outputs, "001");
-  addTableToOutput<Collisions>(outputs, "001");
+  addTableToOutput<Cascades>(outputs, 1);
+  addTableToOutput<Collisions>(outputs, 1);
   addTableToOutput<Decay3Bodys>(outputs);
-  addTableToOutput<FDDs>(outputs, "001");
+  addTableToOutput<FDDs>(outputs, 1);
   addTableToOutput<FT0s>(outputs);
   addTableToOutput<FV0As>(outputs);
   addTableToOutput<StoredFwdTracks>(outputs);
   addTableToOutput<StoredFwdTracksCov>(outputs);
   addTableToOutput<McCollisions>(outputs);
-  addTableToOutput<McCollisionLabels>(outputs);
+  // todo: use addTableToOuput helper?
+  //  currently the description is MCCOLLISLABEL, so
+  //  the name in AO2D would be O2mccollislabel
+  // addTableToOutput<McCollisionLabels>(outputs);
+  outputs.emplace_back(OutputLabel{"McCollisionLabels"}, "AOD", "MCCOLLISIONLABEL", 0, Lifetime::Timeframe);
   addTableToOutput<McMFTTrackLabels>(outputs);
   addTableToOutput<McFwdTrackLabels>(outputs);
-  addTableToOutput<StoredMcParticles_001>(outputs, "001");
+  addTableToOutput<StoredMcParticles_001>(outputs, 1);
   addTableToOutput<McTrackLabels>(outputs);
   addTableToOutput<StoredMFTTracks>(outputs);
   addTableToOutput<StoredTracksIU>(outputs);
@@ -2704,12 +2708,12 @@ DataProcessorSpec getAODProducerWorkflowSpec(GID::mask_t src, bool enableSV, boo
   addTableToOutput<AmbiguousTracks>(outputs);
   addTableToOutput<AmbiguousMFTTracks>(outputs);
   addTableToOutput<AmbiguousFwdTracks>(outputs);
-  addTableToOutput<V0s>(outputs, "001");
-  addTableToOutput<Zdcs>(outputs, "", 1);
+  addTableToOutput<V0s>(outputs, 1);
+  addTableToOutput<Zdcs>(outputs, 1);
   addTableToOutput<Calos>(outputs);
   addTableToOutput<CaloTriggers>(outputs);
   addTableToOutput<CPVClusters>(outputs);
-  addTableToOutput<McCaloLabels>(outputs, "001", 1);
+  addTableToOutput<McCaloLabels_001>(outputs, 1);
   addTableToOutput<Origin>(outputs);
 
   outputs.emplace_back(OutputSpec{"TFN", "TFNumber"});

